### ISSUE TITLE
Secret scrub OTEL logs

### ISSUE THREAD

**Issue by jedevc** [dagger/dagger#8475]
Currently we use the secret scrubber just for the results of `withExec` - this is neccessary so we don't end up accidentally caching secrets in the output.

However, this isn't entirely sufficient - secrets can still appear in the logs, for example by calling `dagger call get-secret plaintext` (where `get-secret` is a field that returns a `Secret`). Since these logs are uploaded to dagger cloud, we should censor all the logs.

This will additionally have the effect of scrubbing the cli output, which is *probably* desirable?

Related:
- https://github.com/dagger/dagger/issues/8464

**Comment by sipsma** [comment]
100% agree with scrubbing this from the OTEL logs and agree with scrubbing it from CLI output *other than* the one case where a user explicitly asks for the CLI to output the plaintext secret (e.g. `dagger call get-secret plaintext`).
* I think we should have an exception for that case since it seems like a valid use case for dagger to support telling users the values of secrets created by modules. E.g. a module that generates certs for you would likely model private keys as Secrets, but it also makes sense to print those out if the user explicitly does `dagger call cert-gen private-key plaintext`

---

Hypothetical implementation notes:
* The client-side doesn't know the values of all secrets created in the session. It *could* know the ones it explicitly created via `SetSecret`, but Secrets can also be created by module function calls, in which case the value is set server-side
* The current best idea I can think of is to create a session attachable that tells the client every secret that gets created in the session. We'd just call this attachable every time a secret is created and the client would store the key/values in an in-memory map.
* Then the client should be able to do scrubbing of any secrets on its side then too (OTEL, terminal output other than the case mentioned above where `plaintext` is explicitly called, etc.)

**Comment by helderco** [comment]
Yeah, I agree we should let the user get the plaintext if explicitly requested in the CLI output, just scrubbed from OTEL. For example you could use a dagger module to get a secret like you can with the `op` CLI.
